### ISSUE TITLE
Animation and effect bugs

### ISSUE THREAD

**Issue by Dexie** [TrinityCore/TrinityCore#8006]
Here is a problem with animation+effect, "Silencing shot".
it have to silence the target immediately, and not after the Animation will come to the target, in arena or BG's its very important, because max. range of hunter is really big range, and till the animation of silencing shot flies to the target, the harmful or the helpful cast is done and it makes the silence to be spent.
Tested on "Arena tournament" PVP is working almost perfect there, you can watch every off/retail movie. I played on other server and found it broken as in the server, the captured video is relevantly to Wowill too, didn't changed the movie.
Same thing is happening with Scatter Shot... 21 yard, but still reaction+flytime= successful cast of enemy.
http://youtu.be/Af9iX48UchE

Same thing is happening with Mage's Arcane Barrage, the dmg cause only after the animation of the spell already flew to the target.

---

Core : f2e92b81a30da8150d99644c4f112bd712dbf28e
TBD: 335.49+
##

**Comment by Keader** [comment]
If this bug is real, please open a new ticket.
Arena Tournament is not a prove
